sdk/identity/identity/src/credentials/chainedTokenCredential.ts

```diff
@@ -56,8 +59,17 @@ export class ChainedTokenCredential implements TokenCredential {
       }
     }
 
-    if (!token && errors.length > 0) {
-      const err = new AggregateAuthenticationError(errors);
+    if (errors.length > 0) {
+      errors.forEach((error) => {
+        if (
```

Comment:
seems like it'd be better to use array.find to locate the first valid element if any